src/main/kotlin/com/exactpro/th2/codec/json/JsonPipelineCodec.kt

```diff
@@ -141,14 +136,10 @@ class JsonPipelineCodec : IPipelineCodec {
     override fun encode(messageGroup: MessageGroup): MessageGroup {
         val messages = messageGroup.messagesList
 
-        if (messages.isEmpty() || messages.none { it.message.metadata.protocol == PROTOCOL }) {
-            return messageGroup
-        }
-
         val builder = MessageGroup.newBuilder()
 
         for (message in messages) {
-            if (message.message.metadata.protocol != PROTOCOL) {
+            if (message.message.metadata.run { protocol.isNotEmpty() && protocol != PROTOCOL }) {
```

Comment:
To have the same logic as in core part we should accept messages with empty protocol only if all parsed messages in a group have empty protocol

Reply:
But the core part has already checked this for us. So this check is okay